.github/scripts/create-file-index.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+
+import os
+import time
+import json
+import sys
+import gzip
+from datetime import datetime
+
+def list_files_with_metadata(directory, output_file):
```

Comment:
write a simple test, could even probably be in this file, where you populate directory with nested folders and symlinks and you know the ground truth to aim for and compare against.